genai-perf/tests/test_json_exporter.py

```diff
@@ -675,10 +679,277 @@ def test_invalid_goodput_json_output(
+                }
+                },
+                "gpu_power_limit": {
+                "unit": "W",
+                "gpu0": {
+                    "avg": 300.0
+                }
+                },
+                "energy_consumption": {
+                "unit": "MJ",
+                "gpu0": {
+                    "avg": 2.154032905081084,
+                    "p25": 2.1533188160000027,
+                    "p50": 2.154057423000003,
+                    "p75": 2.154666464000003,
+                    "p90": 2.155118764000003,
+                    "p95": 2.155270593000003,
+                    "p99": 2.1553677661200026,
+                    "min": 2.1527738520000033,
+                    "max": 2.1554224260000026
+                }
+                },
+                "gpu_utilization": {
+                "unit": "%",
+                "gpu0": {
+                    "avg": 8.72972972972973,
+                    "p25": 9.0,
+                    "p50": 9.0,
+                    "p75": 9.0,
+                    "p90": 10.0,
+                    "p95": 10.0,
+                    "p99": 10.0,
+                    "min": 0.0,
+                    "max": 10.0
+                }
+                },
+                "total_gpu_memory": {
+                "unit": "GB",
+                "gpu0": {
+                    "avg": 51.52702464
+                }
+                },
+                "gpu_memory_used": {
+                "unit": "GB",
+                "gpu0": {
+                    "avg": 26.052919296000006,
+                    "p25": 26.052919296000002,
+                    "p50": 26.052919296000002,
+                    "p75": 26.052919296000002,
+                    "p90": 26.052919296000002,
+                    "p95": 26.052919296000002,
+                    "p99": 26.052919296000002,
+                    "min": 26.052919296000002,
+                    "max": 26.052919296000002
+                }
+                }
+            }
+        """
+
+        cli_cmd = [
+            "genai-perf",
+            "profile",
+            "-m",
+            "gpt2_vllm",
+            "--backend",
+            "vllm",
+            "--streaming",
+            "--extra-inputs",
+            "max_tokens:256",
+            "--extra-inputs",
+            "ignore_eos:true",
+        ]
+        monkeypatch.setattr("sys.argv", cli_cmd)
+        args, _ = parser.parse_args()
+        config = ExporterConfig()
+        config.stats = stats
+        config.telemetry_stats = telemetry_stats
+        config.args = args
+        config.extra_inputs = parser.get_extra_inputs_as_dict(args)
+        config.artifact_dir = args.artifact_dir
+        json_exporter = JsonExporter(config)
+        assert json_exporter._stats_and_args["telemetry_metrics"] == json.loads(
+            expected_telemetry_json_output
+        )
+
+        json_exporter.export()
+        expected_filename = "profile_export_genai_perf.json"
+        written_data = [
```

Comment:
will this ever be more than 1 value?
a list made me think it would be >1 entry but then you assert on 1 entry below.